porter/main.py

```diff
@@ -125,6 +123,22 @@ def __init__(
 
         self.log.info(BANNER)
 
+    @staticmethod
+    def _initialize_endpoints(eth_endpoint: str, polygon_endpoint: str):
+        if not eth_endpoint:
```

Comment:
I guess this is not reachable if Porter is launched via the CLI, but for cases when it's instantiated programatically?

Reply:
Yes. However, perform the None check in the __init__ now along with some other clean up of parameters.